langfuse-core/src/index.ts

```diff
@@ -597,6 +602,54 @@ abstract class LangfuseCoreStateless {
     }
   }
 
+  /**
+   * Truncates the event body if its byte size exceeds the specified maximum byte size.
+   * Emits a warning event if truncation occurs.
+   * The fields that may be truncated are: "input", "output", and "metadata".
+   * The fields are truncated in the order of their size, from largest to smallest until the total byte size is within the limit.
+   */
+  protected truncateEventBody(body: EventBody, maxByteSize: number): EventBody {
+    const bodySize = this.getByteSize(body);
+
+    if (bodySize <= maxByteSize) {
+      return body;
+    }
+
+    this._events.emit("warning", `Event Body is too large (${bodySize} bytes) and will be truncated`);
+
+    // Sort keys by size and truncate the largest keys first
+    const keysToCheck = ["input", "output", "metadata"] as const;
+    const keySizes = keysToCheck
+      .map((key) => ({ key, size: key in body ? this.getByteSize(body[key as keyof typeof body]) : 0 }))
+      .sort((a, b) => b.size - a.size);
+
+    let result = { ...body };
+    let currentSize = bodySize;
+
+    for (const { key, size } of keySizes) {
+      if (currentSize > maxByteSize && Object.prototype.hasOwnProperty.call(result, key)) {
+        result = { ...result, [key]: "<truncated due to size exceeding limit>" };
+
+        this._events.emit("warning", `Truncated ${key} due to total size exceeding limit`);
+
+        currentSize -= size;
+      }
+    }
+
+    return result;
+  }
+
+  private getByteSize(obj: any): number {
+    const serialized = JSON.stringify(obj);
+
+    // Use TextEncoder if available, otherwise fallback to encodeURIComponent
+    if (typeof TextEncoder !== "undefined") {
+      return new TextEncoder().encode(serialized).length;
+    } else {
+      return encodeURIComponent(serialized).replace(/%[A-F\d]{2}/g, "U").length;
+    }
```

Comment:
Is there any environment (e.g. CF workers) where this is failing? I assume this is exactly what you do with the above.

Reply:
e.g. TextEncoder is available on Cloudflare workers as well https://developers.cloudflare.com/workers/runtime-apis/encoding/